Network/SerializationHooks.cs

```diff
@@ -37,57 +37,52 @@ public static void Uninitialize()
         _deserializeDetour?.Dispose();
     }
 
+
     [UnmanagedFunctionPointer(CallingConvention.StdCall)]
-    public unsafe delegate void SerializeEvent(IntPtr entityManager, UInt64 networkEventType, UInt64 netBufferOut, IntPtr entity);
+    public unsafe delegate void SerializeEvent(IntPtr entityManager, IntPtr networkEventType, IntPtr netBufferOut, IntPtr entity);
 
     public static SerializeEvent? SerializeOriginal;
 
-    public unsafe static void SerializeHook(IntPtr entityManager, UInt64 networkEventType, UInt64 netBufferOut, IntPtr entity)
+    public unsafe static void SerializeHook(IntPtr entityManager, IntPtr networkEventType, IntPtr netBuffer, IntPtr entity)
```

Comment:
Taking these parameters as native integers is redundant, you end up turning them into the relevant ref structs anyway. Take them in as ref parameters instead.

```suggestion
    public unsafe static void SerializeHook(IntPtr entityManager, IntPtr networkEventType, ref NetBufferOut netBuffer, ref Entity entity)
```